assertj-core/src/main/java/org/assertj/core/api/recursive/comparison/FieldLocation.java

```diff
@@ -33,7 +36,7 @@ public final class FieldLocation implements Comparable<FieldLocation> {
 
   private final String pathToUseInRules;
   private final List<String> decomposedPath;
-  private final List<String> pathsHierarchyToUseInRules;
+  private final Set<String> pathsHierarchyToUseInRules;
```

Comment:
provide `O(1)` `contains` instead of `O(N)` for `List#contains` used by `hierarchyMatches(String)`